pkgs/development/libraries/boost/generic.nix

```diff
@@ -114,7 +114,8 @@ stdenv.mkDerivation {
     then ./darwin-1.55-no-system-python.patch
     else ./darwin-no-system-python.patch)
   ++ optional (and (versionAtLeast version "1.70") (!versionAtLeast version "1.73")) ./cmake-paths.patch
-  ++ optional (versionAtLeast version "1.73") ./cmake-paths-173.patch;
+  ++ optional (versionAtLeast version "1.73") ./cmake-paths-173.patch
+  ++ optional (stdenv.hostPlatform != stdenv.buildPlatform) ./cross.patch;
```

Comment:
This patch doesn't apply to boost 1.75. After fixing up the patch, I was able to cross-compile to riscv64.

IMO patches should be applied unconditionally whenever possible. If they're conditional on certain architectures or setups, they can break silently after version bumps. In this case, that means a mass rebuild, so we'll want to send this through staging.

Reply:
Can you post the correct patch, please?
Or push it here or open a new PR.

Reply:
Heads up that this probably won't apply to all versions, I only tested this on 1.75.

```diff
index ca0b08d58..5d4105ffe 100755
--- a/bootstrap.sh
+++ b/bootstrap.sh
@@ -225,4 +225,4 @@
   pwd=`pwd`
-  (cd "$my_dir/tools/build/src/engine" && ./build.sh)
+  (cd "$my_dir/tools/build/src/engine" && CXX="$CXX_FOR_BUILD" ./build.sh)
   if [ $? -ne 0 ]; then
```